src/libcore/ops.rs

```diff
@@ -904,8 +919,29 @@ impl<Idx: Clone + Step> DoubleEndedIterator for Range<Idx> {
 #[unstable = "API still in development"]
 impl<Idx: Clone + Step> ExactSizeIterator for Range<Idx> {}
 
+#[unstable = "API still in development"]
+impl<Idx: fmt::Show> fmt::Show for Range<Idx> {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        write!(fmt, "{:?}..{:?}", self.start, self.end)
+    }
+}
+#[cfg(stage0)]
+#[unstable = "API still in development"]
+impl<Idx: fmt::String + fmt::Show> fmt::String for Range<Idx> {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        write!(fmt, "{}..{}", self.start, self.end)
+    }
+}
+#[cfg(not(stage0))]
+#[unstable = "API still in development"]
+impl<Idx: fmt::String> fmt::String for Range<Idx> {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        write!(fmt, "{}..{}", self.start, self.end)
+    }
+}
```

Comment:
Could the `String` implementations be left out for now, and just have `Show` implementations provided?

(still working out precisely what `String` is for and for not)